File: python/src/thunder_book/ch05/random_action.py
import random

import fire

from thunder_book.ch05.maze_state import AlternateMazeState, MazeParams


def random_action(state: AlternateMazeState) -> int:
    legal_actions = state.legal_actions()
    return random.choice(legal_actions)


def play_game(seed: int = 0) -> None:
    state = AlternateMazeState(seed=seed, params=MazeParams(height=3, width=4, end_turn=10))
    print(state)
    p = 0
    while not state.is_done():
        print(f"{p+1}p {'-'*20}")
        action = random_action(state)
        print(f"action:\t{action}")
        state.advance(action)
        print(state)
        p ^= 1
    state.print_end_game()


if __name__ == "__main__":
    fire.Fire(play_game)

File: cpp/src/ch06/random_action.cc
#include <random>
#include "random_action.h"

auto mt_for_action = std::mt19937(0);

int random_action(const State &state, const int player_id)
{
    auto legal_actions = state.legal_actions(player_id);
    return legal_actions[mt_for_action() % (legal_actions.size())];
}
